cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -138,6 +141,9 @@ def index(cls, modulestore, structure_key, triggered_at=None, reindex_age=REINDE
         if not searcher:
             return
 
+        if hasattr(searcher, "update_settings"):
+            searcher.update_settings()
```

Comment:
1. `update_settings` is not a clear name. I would prefer something like `ensure_index_exists()` or `prepare_index()`
2. You should define and document this on the base class, so we can call it unconditionally here. Then it's part of the abstract API; but here it's just Meilisearch-specific code.